### PR TITLE
Fix tests for concurrent search with IndexRandomForConcurrentSearch f…

### PR DESCRIPTION
**Pull request by Gankris96**
follow-up [#11130]

### Description
This fix follows up the pr - [#11130] with few additional fixes 
This pr addresses the following tests 

```

SearchScrollIT
.testSimpleScrollQueryThenFetch
.testSimpleScrollQueryThenFetchSmallSizeUnevenDistribution
.testScrollAndUpdateIndex
.testSimpleScrollQueryThenFetch_clearScrollIds
.testSimpleScrollQueryThenFetchClearAllScrollIds
.testDeepScrollingDoesNotBlowUp
.testStringSortMissingAscTerminates
.testCloseAndReopenOrDeleteWithActiveScroll
.testScrollInvalidDefaultKeepAlive
.testRestartDataNodesDuringScrollSearch


FieldSortIT
.testIssue8226
.testScoreSortDirection
.testScoreSortDirectionWithFunctionScore
.testIssue2991
.testSimpleSorts
.testSortMissingNumbers
.testSortMissingNumbersMinMax
.testSortMissingStrings
.testSortMVField
.testSortOnRareField
.testSortMetaField
.testNestedSort
.testFieldAlias
.testFieldAliasesWithMissingValues
.testCastDate
.testLongSortOptimizationCorrectResults



GeoDistanceIT
.testDistanceSortingMVFields
.testDistanceSortingWithMissingGeoPoint
.testDistanceSortingWithUnmappedField

SourceFetchingIT
.testSourceFiltering
.testSourceWithWildcardFiltering


SuggestSearchIT
.testSimple

```

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

https://github.com/opensearch-project/OpenSearch/pull/10247

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] ~~New functionality has been documented.~~
  - [ ] ~~New functionality has javadoc added~~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [ ] ~~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~~
- [ ] ~~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jed326 — APPROVED** [review]

File: server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollIT.java [hunk 1]
```diff
@@ -135,6 +135,7 @@ public void testSimpleScrollQueryThenFetch() throws Exception {
         }
 
         client().admin().indices().prepareRefresh().get();
+        indexRandomForConcurrentSearch("test");
```
Comment (jed326) [inline server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollIT.java:138]: A few of these tests, like this one, look like the index many documents already. For those that are already indexing a lot of documents do we still need to add this?

Comment (Gankris96) [reply server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollIT.java:138]: it doesn't go down the 2 slice path if we dont add indexRandomForConcurrentSearch call.

**Comment by sohami** [comment]
@Gankris96 precommit and assemble are failing, can you please take a look ?

**Review by sohami — APPROVED** [review]

**Comment by sohami** [comment]
> ```
> SearchWeightedRoutingIT.testMultiGetWithNetworkDisruption_FailOpenEnabled
> ```

Existing one: https://github.com/opensearch-project/OpenSearch/issues/10755

**Pull request merged**